bug: claude-mem worker startup race condition on port 37777

## Problem

The claude-mem worker fails to start on the first attempt due to a port 37777 conflict, then auto-recovers after killing orphaned processes:

```
[2026-03-27 18:29:54.028] [ERROR] [SYSTEM] Worker not available {}
[2026-03-27 18:29:54.028] [ERROR] [SYSTEM] Tools will fail until Worker is started
[2026-03-27 18:29:54.028] [ERROR] [SYSTEM] Start Worker with: npm run worker:restart
[2026-03-27 18:29:55.445] [ERROR] [SYSTEM] ✗ Worker failed to start Failed to start server. Is port 37777 in use?
```

The worker eventually recovers (~25s later) after cleanup:

```
[2026-03-27 18:30:19.577] Worker became healthy
```

## Details

- **Component**: claude-mem search server worker
- **Port**: 37777
- **Recovery time**: ~25 seconds
- **Log file**: `~/.claude-mem/logs/claude-mem-2026-03-27.log`

## Root Cause

During container startup, there appears to be a race condition where:
1. A previous worker process (or orphaned child) still holds port 37777
2. The new worker attempts to bind the same port and fails
3. The cleanup mechanism kills 6 orphaned processes and retries successfully

This likely happens because the SessionStart hook triggers the worker before the previous instance from entrypoint.sh has fully released the port.

## Additional Warnings

The claude-mem log also shows recurring parser warnings:
```
[WARN] [PARSER] [session-1] Summary response contained <observation> tags instead of <summary> — prompt conditioning may need strengthening
```
This occurs 5+ times per session and indicates the LLM response format doesn't match what the parser expects.

## Impact

- **Severity**: Low — auto-recovery works, but tools fail for the first ~25s of a session
- The parser warnings are cosmetic but indicate potential prompt drift

## Suggested Fix

1. Add a port availability check before starting the worker
2. Use `SO_REUSEADDR` or `SO_REUSEPORT` socket option on the worker
3. Add a pre-start cleanup step that kills any existing process on port 37777
4. For the parser warning: upstream issue with claude-mem prompt conditioning

## Environment

- **Build**: `da299c13` (2026-03-27)
- **claude-mem**: 10.6.2

--- a/claude-config/chrome-browser.sh
+++ b/claude-config/chrome-browser.sh
@@ -61,6 +61,7 @@ start_chrome_browser() {
     --disable-background-timer-throttling
     --disable-backgrounding-occluded-windows
     --disable-dev-shm-usage
+    --disable-background-networking
   )
 
   # Conditional GPU support

--- a/claude-config/claude.json
+++ b/claude-config/claude.json
@@ -14,7 +14,10 @@
   "mcpServers": {
     "chrome-devtools": {
       "command": "npx",
-      "args": ["-y", "chrome-devtools-mcp@0.20.2", "--browserUrl=http://localhost:9222"]
+      "args": ["-y", "chrome-devtools-mcp@0.20.2", "--browserUrl=http://localhost:9222"],
+      "env": {
+        "NODE_OPTIONS": "--no-warnings"
+      }
     }
   },
   "projects": {

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -41,6 +41,9 @@ if [ -n "$GH_TOKEN" ]; then
   gh auth setup-git 2>/dev/null || true
 fi
 
+# Ensure Homebrew npm global directory exists (issue #677)
+mkdir -p "${HOME}/.npm-global/lib" 2>/dev/null || true
+
 # ============================================================
 # gogcli (gog) — restore OAuth credentials and refresh token
 # ============================================================
@@ -220,6 +223,15 @@ if [ "${ENABLE_VNC:-false}" = "true" ]; then
     fi
 
   ) &
+fi
+
+# ============================================================
+# Clean up stale processes on claude-mem worker port (issue #676)
+# Prevents port 37777 conflict when SessionStart hook triggers
+# the claude-mem worker before a previous instance has released.
+# ============================================================
+if command -v fuser >/dev/null 2>&1; then
+  fuser -k 37777/tcp >/dev/null 2>&1 || true
 fi
 
 # ============================================================

